### PR TITLE
feat: added limited storage class support 

### PR DESCRIPTION
**Pull request by 4o4x**
This pull request adds support for storageClass to the s3fake project. 

Basic scenarios related to storage classes have been tested.

### CODE REVIEW

File: backend.go [hunk 1]
```diff
@@ -226,7 +227,7 @@ type Backend interface {
 	//
 	// The size can be used if the backend needs to read the whole reader; use
 	// gofakes3.ReadAll() for this job rather than ioutil.ReadAll().
-	PutObject(bucketName, key string, meta map[string]string, input io.Reader, size int64) (PutObjectResult, error)
+	PutObject(bucketName, key string, meta map[string]string, input io.Reader, size int64, storageClass string) (PutObjectResult, error)
```
Comment (igungor) [inline backend.go]: You may use the `StorageClass` type here.

File: gofakes3.go [hunk 2]
```diff
@@ -618,7 +618,15 @@ func (g *GoFakeS3) createObject(bucket, object string, w http.ResponseWriter, r
 		return err
 	}
 
-	result, err := g.storage.PutObject(bucket, object, meta, rdr, size)
+	if r.Header.Get("x-amz-storage-class") != "" {
+		if ok := IsStorageClassValid(r.Header.Get("x-amz-storage-class")); !ok {
+			return ErrInvalidStorageClass
+		}
+	} else {
```
Comment (igungor) [inline gofakes3.go]: Why do you validate the storage class? There're several s3-compatible object storage services that have diffent storage class names.

Comment (4o4x) [reply gofakes3.go]: done.

File: backend.go (hunk 1 above)
Comment (4o4x) [reply backend.go]: done.

**Comment by igungor** [comment]
Pushed v0.0.17.

**Pull request merged**